org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -851,4 +851,21 @@ public static void setCompatibleVMs(String id) {
 			}
 		}
 	}
+
+	public static IResource getFileOrFolder(String uriString) {
+		IFile file = findFile(uriString); // This may return IFile even when uriString really describes a IContainer
+		IContainer parent = file == null ? null : file.getParent();
+		if (parent == null) {
+			return file;
+		}
+		try {
+			parent.refreshLocal(DEPTH_ONE, null);

```

Comment:
so if file is an actual file, then we're, refreshing its parent folder (and the file siblings). Seems suboptimal to me